app/src/main/java/edu/uc/coffeens/stocktracker/ui/main/MainViewModel.kt

```diff
@@ -31,15 +31,18 @@ class MainViewModel : ViewModel() {
         stock = stockService.fetchStock()
     }
 
-    fun save(stocks: Stock) {
+    /**
+     * TODO: Implement data save function
+     */
+    fun save(stock: Stock) {
         firestore.collection("Stocks")
             .document()
-            .set(stocks)
+            .set(stock)
             .addOnSuccessListener {
-                Log.d("Firebase", "document saved")
+                Log.i("Firebase", "document saved")
             }
             .addOnFailureListener {
-                Log.d("Firebase", "save failed")
+                Log.e("Firebase", "save failed")
```

Comment:
Log.e for ERROR https://developer.android.com/reference/kotlin/android/util/Log

Reply:
Looks good.